src/azure-cli/azure/cli/command_modules/monitor/operations/log_alert.py

```diff
@@ -0,0 +1,252 @@
+                    (x for x in instance.action.azns_action.action_group if action_group.lower() == x.lower()), None
+                )
+                if not match:
+                    instance.action.azns_action.action_group.append(action_group)
+
+    if remove_action_groups:
+        remove_action_groups = _normalize_names(cmd.cli_ctx, remove_action_groups, resource_group, 'microsoft.insights',
+                                                'actionGroups')
+        from knack.util import CLIError
+        if instance.action.azns_action.action_group is None:
+            raise CLIError('Error in removing action group. There are no action groups attached to alert rule.')
+
+        for action_group in remove_action_groups:
+            match = next(
+                (x for x in instance.action.azns_action.action_group if action_group.lower() == x.lower()), None
+            )
+            if match:
+                instance.action.azns_action.action_group.remove(action_group)
+            else:
+                raise CLIError(
+                    'Error in removing action group. Action group "{}" is not attached to alert rule.'
+                    .format(action_group))
+
+    return instance
+
+
+def update_authorized_resources(instance, reset_authorized_resources=None, add_authorized_resources=None,
+                                remove_authorized_resources=None):
+    if reset_authorized_resources:
+        instance.source.authorized_resources = None
+
+    if add_authorized_resources:
+        if instance.source.authorized_resources is None:
+            instance.source.authorized_resources = add_authorized_resources
+        else:
+            for authorized_resources in add_authorized_resources:
+                match = next(
+                    (x for x in instance.source.authorized_resources if authorized_resources.lower() == x.lower()), None
+                )
+                if not match:
+                    instance.source.authorized_resources.append(authorized_resources)
+
+    if remove_authorized_resources:
+        from knack.util import CLIError
+        if instance.source.authorized_resources is None:
+            raise CLIError(
+                'Error in removing authorized resource. There are no authorized resources attached to alert rule.')
+
+        for authorized_resources in remove_authorized_resources:
+            match = next(
+                (x for x in instance.source.authorized_resources if authorized_resources.lower() == x.lower()), None
+            )
+            if match:
+                instance.source.authorized_resources.remove(authorized_resources)
+            else:
+                raise CLIError(
+                    'Error in removing authorized resource. Authorized resource "{}" is not attached to alert rule.'
+                    .format(authorized_resources))
+
+    return instance
+
+
+def _normalize_names(cli_ctx, resource_names, resource_group, namespace, resource_type):
```

Comment:
you could remove this function to the _validate.py and use validator=validate_action_group in the _param.py

Reply:
How can I get subscription id inside the validator function?
Currently we do this:
subscription=get_subscription_id(cli_ctx)

Reply:
subscription=get_subscription_id(cmd.cli_ctx),